src/authenticator_proxy.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const AWSXRayMock = require("./aws_xray_mock");
+
+class AuthenticatorProxy {
+
+  constructor(authenticator, xray) {
+    this.authenticator = authenticator ? authenticator : { getAuthorization: () => Promise.resolve() };
+    this.xray = xray ? xray : AWSXRayMock;
+  }
+
+  /**
+   * @param {*} segmentName String, // X-Ray segment name
+   * @param {*} promise Wrapped function
+   * @param {*} annotations  Array<Object> // Annotations to be included, {key, value}
+   */
+  xRayCapture(segmentName, promise, annotations) {
```

Comment:
Is this wrapping the authenticator, or is this wrapping the underlying call in the promise parameter ? This naming is a little confusing to me because reading through the code, the name didn't really tell me much - I had to read all the code to figure out how this is used

Reply:
Let me improve the naming and description.